index.js

```diff
@@ -9,42 +9,65 @@ var hasFlag = function (flag) {
 };
 
 module.exports = (function () {
-	if ('FORCE_COLOR' in process.env) {
-		return true;
+	var support = function (level) {
+		if (level <= 0) return null;
+		var result = {};
+		result.level = level;
+		result.has16 = result.hasBasic = level >= 1;
+		result.has256 = level >= 2;
+		result.has16m = level >= 3;
+		return result;
 	}
 
-	if (hasFlag('no-color') ||
-		hasFlag('no-colors') ||
-		hasFlag('color=false')) {
-		return false;
-	}
+	var result = (function () {
```

Comment:
This can be made into a normal function and put in the top scope. It can be called and cached in the IIFE anyways.